jenkins/Jenkinsfile

```diff
@@ -1,374 +1,492 @@
+                            sh """
+                                sed -i "s/$image.*/$image:$version\\"/" jenkins/gce-custom-images/prepare_custom_leonardo_gce_image.sh
+                            """
+                        }
+
+                        versionMap[image] = version
+                    }
+
+                    if (GENERATE_DATAPROC_IMAGE == "true") {
+                        sh """
+                            printf 'Printing any changed versions in prepare-custom-leonardo-jupyter-dataproc-image.sh:\\n\\n'
+                            git diff jenkins/dataproc-custom-images/prepare-custom-leonardo-jupyter-dataproc-image.sh
+                        """
+                    }
+
+                    if (GENERATE_GCE_IMAGE == "true") {
+                        sh """
+                            printf 'Printing any changed versions in prepare_custom_leonardo_gce_image.sh:\\n\\n'
+                            git diff jenkins/gce-custom-images/prepare_custom_leonardo_gce_image.sh
+                        """
+                    }
+                }
+            }
     }
 
-      stage('Run build script') {
+    stage('Run image generation scripts') {
         when { expression { !shouldAbortBuild } }
-        steps {
-          sh """
-              git submodule init && git submodule update
-          """
-          script {
-            for(String dp_version: dataproc_versions.trim().split(' ')) {
-              def dp_version_formatted = dp_version.replace(".","-")
-              // Dataproc 1.2 is used for the legacy image and must have python 3.6 installed
-              // Otherwise, a terra-docker image is used which must have python 3.7 installed
-              def python_version = dp_version.startsWith("1.2") ? "3.6.8" : "3.7.4"
-              println("Installing Python $python_version on Dataproc $dp_version")
-              sh (
-                script: """
-                  sed -i 's/python_version=.*/python_version="$python_version"/' jenkins/dataproc-custom-images/prepare-custom-leonardo-jupyter-dataproc-image.sh
-                  cat jenkins/dataproc-custom-images/prepare-custom-leonardo-jupyter-dataproc-image.sh
-                  cd jenkins/dataproc-custom-images/dataproc-custom-images
-                  python generate_custom_image.py \
-                    --image-name "$IMAGE_BASE_NAME-$dp_version_formatted-$imageID" \
-                    --dataproc-version "$dp_version" \
-                    --customization-script ../prepare-custom-leonardo-jupyter-dataproc-image.sh \
-                    --zone us-central1-a \
-                    --gcs-bucket $BUCKET_NAME \
-                    --project-id=$GOOGLE_PROJECT \
-                    --disk-size=50
-                """
-              )
+            // Fail the pipeline if any of the parallel branches (that build the Dataproc and GCE images) fails
+            failFast true
```

Comment:
why do we want to fail the other custom image build if one fails? Sometimes these builds can take a while, itsay it may be worth setting this to false, since it will require manual intervention to build the other anyways

Reply:
I thought a failure in one might be indicative of something wrong in the other one as well in which case I figured we wouldn't err on the side of caution and not merge. If you think that concern is not to optimize for, I'd be happy to set `failFast` to false since you have a lot more experience with these builds.